internal/cmd/server/backup-schedule/delete/delete.go

```diff
@@ -0,0 +1,118 @@
+package delete
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/stackitcloud/stackit-cli/internal/pkg/args"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/errors"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/examples"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/flags"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/globalflags"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/print"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/services/serverbackup/client"
+	serverbackupUtils "github.com/stackitcloud/stackit-cli/internal/pkg/services/serverbackup/utils"
+
+	"github.com/spf13/cobra"
+	"github.com/stackitcloud/stackit-sdk-go/services/serverbackup"
+)
+
+const (
+	scheduleIdArg = "SCHEDULE_ID"
+	serverIdFlag  = "server-id"
+)
+
+type inputModel struct {
+	*globalflags.GlobalFlagModel
+	ScheduleId string
+	ServerId   string
+}
+
+func NewCmd(p *print.Printer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("delete %s", scheduleIdArg),
+		Short: "Deletes a Server Backup Schedule",
+		Long:  "Deletes a Server Backup Schedule.",
+		Args:  args.SingleArg(scheduleIdArg, serverbackupUtils.ValidBackupScheduleIdString),
+		Example: examples.Build(
+			examples.NewExample(
+				`Delete a Server Backup Schedule with ID "xxx" for server "zzz"`,
+				"$ stackit server backup-schedule delete --server-id=zzz xxx"),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+			model, err := parseInput(p, cmd, args)
+			if err != nil {
+				return err
+			}
+
+			// Configure API client
+			apiClient, err := client.ConfigureClient(p)
+			if err != nil {
+				return err
+			}
+
+			if !model.AssumeYes {
+				prompt := fmt.Sprintf("Are you sure you want to delete server backup schedule %q? (This cannot be undone)", model.ScheduleId)
+				err = p.PromptForConfirmation(prompt)
+				if err != nil {
+					return err
+				}
+			}
+
+			// Call API
+			req := buildRequest(ctx, model, apiClient)
+			err = req.Execute()
+			if err != nil {
+				return fmt.Errorf("delete Server Backup Schedule: %w", err)
+			}
+
+			operationState := "Deleted"
+			if model.Async {
+				operationState = "Triggered deletion of"
+			}
+			p.Info("%s server backup schedule %q\n", operationState, model.ScheduleId)
```

Comment:
Is any of these operations async? Meaning that they return before finishing and then we need to check it's status through either a status field or until the get returns a 404?

Reply:
If not then you do not need the `Triggered ...` part

Reply:
This also applies for the `create` and `update` commands